flang/test/Fir/box.fir

```diff
@@ -1,5 +1,23 @@
 // RUN: tco -o - %s | FileCheck %s
 
+// Global box initialization (test must come first because llvm globals are emitted first).
+// CHECK-LABEL: @globalx = internal global { i32*, i64, i32, i8, i8, i8, i8 } { i32* null, i64 4, i32 20180515, i8 0, i8 9, i8 2, i8 0 }
+fir.global internal @globalx : !fir.box<!fir.heap<i32>> {
```

Comment:
I have some concerns about what looks like a type system change here...

Reply:
What type would you expect here for `@globalx` ?
Because it is a scalar, I agree it could be a simple `!fir.heap<i32>`, but this is not true for allocatable arrays, or scalar derived-type/characters where we need to store more than just the base pointer. So my approach choice here is motivated by having a uniform handling of allocatables in lowering.

Reply:
I was expecting the type `!fir.box<i32>` for `@globalx` (per other discussion on CFI_attribute_t, etc.).
You might be able to save the std.constant Op here with
```mlir
%0 = fir.undefined !fir.heap<i32>
```

Reply:
Does `fir.undefined` guarantees that it will be set to zero ?
Here the value must be set to zero because we want `ALLOCATED()` to answer false on the initial descriptor value.
Quickly going through llvm doc regarding UndefOp, I read "Undefined values are useful because they indicate to the compiler that the program is well defined no matter what value is used.". Is `fir.undefined` semantics a bit different here ?